File: Data Structures & Algorithms/validate-parentheses/submission-2.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []

        matches = {']':'[', '}':'{', ')':'(' }

        for i in s:
            if stack and i in matches.keys():
                if stack[-1] != matches[i]:
                    return False
                else:
                    stack.pop()
            else:
                stack.append(i)        
        
        return len(stack) == 0

File: Data Structures & Algorithms/invert-a-binary-tree/submission-2.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public TreeNode invertTree(TreeNode root) {

        if (root == null) return null;

        Stack<TreeNode> myStack = new Stack<>();
        
        myStack.push(root);

        while (!myStack.isEmpty())
        {
            TreeNode top = myStack.pop();

            TreeNode left = top.left;
            top.left = top.right;
            top.right = left;

            if (top.right != null) myStack.push(top.right);
            if (top.left != null) myStack.push(top.left);
        }

        return root;
    }
}
